pallets/proposals/src/tests.rs

```diff
@@ -1415,12 +1424,94 @@ fn test_withdraw_upon_project_approval_and_finalised_voting() {
         )
         .unwrap();
 
-        Proposals::finalise_milestone_voting(
+        // Proposals::finalise_milestone_voting(
+        //     RuntimeOrigin::signed(*ALICE),
+        //     project_key,
+        //     milestone1_key,
+        // )
+        // .unwrap();
+
+        assert_ok!(Proposals::withdraw(
             RuntimeOrigin::signed(*ALICE),
+            project_key
+        ));
+
+        assert_eq!(
+            Tokens::free_balance(CurrencyId::Native, &*ALICE),
+            initial_balance + required_funds
+        );
+        let latest_event = <frame_system::Pallet<Test>>::events()
+            .pop()
+            .expect("Expected at least one RuntimeEventRecord to be found")
+            .event;
+        assert_eq!(
+            latest_event,
+            mock::RuntimeEvent::from(proposals::Event::ProjectFundsWithdrawn(
+                *ALICE,
+                0,
+                100,
+                CurrencyId::Native
+            ))
+        );
+    });
+}
+
+#[test]
+fn test_withdraw_upon_project_auto_approval_and_based_on_threshold_met_during_voting() {
```

Comment:
Do we have a negative case for this? checking that it doesnt auto approve if the threshold is not met is important.